### ISSUE TITLE
error importing `is_compiled_module`

### ISSUE THREAD

**Issue by 5shekel** [tencent-ailab/IP-Adapter#73]
https://github.com/tencent-ailab/IP-Adapter/blob/22da45667898fd237ab54d3681db53a9ae98bf1e/ip_adapter/utils.py#L9

after
```python
pip install diffusers[torch]
#or
pip install diffusers
```

im getting
```python
>>> from diffusers.utils import is_compiled_module
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'is_compiled_module' from 'diffusers.utils' (C:\tools\miniconda3\envs\test\lib\site-packages\diffusers\utils\__init__.py)
```

simple `print(dir(diffusers.utils))` cant find it.
```python
>>> import diffusers.utils 
>>> print(dir(diffusers.utils))
['BACKENDS_MAPPING', 'BaseOutput', 'CONFIG_NAME', 'DEPRECATED_REVISION_ARGS', 'DIFFUSERS_CACHE', 'DIFFUSERS_DYNAMIC_MODULE_NAME', 'DummyObject', 'ENV_VARS_TRUE_AND_AUTO_VALUES', 'ENV_VARS_TRUE_VALUES', 'FLAX_WEIGHTS_NAME', 'HF_HUB_OFFLINE', 'HF_MODULES_CACHE', 'HUGGINGFACE_CO_RESOLVE_ENDPOINT', 'ONNX_EXTERNAL_WEIGHTS_NAME', 'ONNX_WEIGHTS_NAME', 'OptionalDependencyNotAvailable', 'PIL_INTERPOLATION', 'PushToHubMixin', 'SAFETENSORS_WEIGHTS_NAME', 'USE_JAX', 'USE_TF', 'USE_TORCH', 'WEIGHTS_NAME', '_LazyModule', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '__version__', '_add_variant', '_get_model_file', 'check_min_version', 'constants', 'deprecate', 'deprecation_utils', 'doc_utils', 'dummy_flax_and_transformers_objects', 'dummy_flax_objects', 'dummy_note_seq_objects', 'dummy_onnx_objects', 'dummy_torch_and_librosa_objects', 'dummy_torch_and_scipy_objects', 'dummy_torch_and_torchsde_objects', 'dummy_torch_and_transformers_and_k_diffusion_objects', 'dummy_torch_and_transformers_and_onnx_objects', 'dummy_torch_and_transformers_objects', 'dummy_transformers_and_torch_and_note_seq_objects', 'dynamic_modules_utils', 'export_to_gif', 'export_to_obj', 'export_to_ply', 'export_to_video', 'export_utils', 'extract_commit_hash', 'get_class_from_dynamic_module', 'get_logger', 'get_objects_from_module', 'http_user_agent', 'hub_utils', 'import_utils', 'is_accelerate_available', 'is_accelerate_version', 'is_bs4_available', 'is_flax_available', 'is_ftfy_available', 'is_inflect_available', 'is_invisible_watermark_available', 'is_k_diffusion_available', 'is_k_diffusion_version', 'is_librosa_available', 'is_note_seq_available', 'is_omegaconf_available', 'is_onnx_available', 'is_scipy_available', 'is_tensorboard_available', 'is_torch_available', 'is_torch_version', 'is_torchsde_available', 'is_transformers_available', 'is_transformers_version', 'is_unidecode_available', 'is_wandb_available', 'is_xformers_available', 'load_image', 'loading_utils', 'logger', 'logging', 'make_image_grid', 'numpy_to_pil', 'os', 'outputs', 'pil_utils', 'pt_to_pil', 'replace_example_docstring', 'requires_backends', 'version']
```
i see others using
```python
from diffusers.utils.torch_utils import is_compiled_module
```

**Comment by xiaohu2015** [comment]
@5shekel hi, it seems that the new version of diffusers has do some changes.

you can use the new code, we refactor the code to handle this.

**Comment by TheLearner23** [comment]
> @5shekel hi, it seems that the new version of diffusers has do some changes.
> 
> you can use the new code, we refactor the code to handle this.

Which version are you referring to? I have installed from source ("0.22.0.dev0" version) and dir(diffusers.utils)  only lists the following. Could you please advise?!

['BACKENDS_MAPPING', 'BaseOutput', 'CONFIG_NAME', 'DEPRECATED_REVISION_ARGS', 'DIFFUSERS_CACHE', 'DIFFUSERS_DYNAMIC_MODULE_NAME', 'DummyObject', 'ENV_VARS_TRUE_AND_AUTO_VALUES', 'ENV_VARS_TRUE_VALUES', 'FLAX_WEIGHTS_NAME', 'HF_HUB_OFFLINE', 'HF_MODULES_CACHE', 'HUGGINGFACE_CO_RESOLVE_ENDPOINT', 'ONNX_EXTERNAL_WEIGHTS_NAME', 'ONNX_WEIGHTS_NAME', 'OptionalDependencyNotAvailable', 'PIL_INTERPOLATION', 'PushToHubMixin', 'SAFETENSORS_WEIGHTS_NAME', 'USE_JAX', 'USE_TF', 'USE_TORCH', 'WEIGHTS_NAME', '_LazyModule', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '__version__', '_add_variant', '_get_model_file', 'check_min_version', 'constants', 'convert_state_dict_to_diffusers', 'convert_state_dict_to_peft', 'deprecate', 'deprecation_utils', 'doc_utils', 'dummy_flax_and_transformers_objects', 'dummy_flax_objects', 'dummy_note_seq_objects', 'dummy_onnx_objects', 'dummy_torch_and_librosa_objects', 'dummy_torch_and_torchsde_objects', 'dummy_torch_and_transformers_and_k_diffusion_objects', 'dummy_torch_and_transformers_and_onnx_objects', 'dummy_transformers_and_torch_and_note_seq_objects', 'dynamic_modules_utils', 'export_to_gif', 'export_to_obj', 'export_to_ply', 'export_to_video', 'export_utils', 'extract_commit_hash', 'get_class_from_dynamic_module', 'get_logger', 'get_objects_from_module', 'http_user_agent', 'hub_utils', 'import_utils', 'is_accelerate_available', 'is_accelerate_version', 'is_bs4_available', 'is_flax_available', 'is_ftfy_available', 'is_inflect_available', 'is_invisible_watermark_available', 'is_k_diffusion_available', 'is_k_diffusion_version', 'is_librosa_available', 'is_note_seq_available', 'is_omegaconf_available', 'is_onnx_available', 'is_peft_available', 'is_scipy_available', 'is_tensorboard_available', 'is_torch_available', 'is_torch_version', 'is_torchsde_available', 'is_transformers_available', 'is_transformers_version', 'is_unidecode_available', 'is_wandb_available', 'is_xformers_available', 'load_image', 'loading_utils', 'logger', 'logging', 'make_image_grid', 'numpy_to_pil', 'os', 'outputs', 'peft_utils', 'pil_utils', 'pt_to_pil', 'recurse_remove_peft_layers', 'replace_example_docstring', 'requires_backends', 'state_dict_utils', 'version']

**Comment by Sivll** [comment]
hello! Do you have any news? still doesn't want to import is_compiled_module from anywhere...(((
python 3.11.3

ImportError: cannot import name 'is_compiled_module' from 'diffusers.utils' (/Users/.../sd/MetalDiffusion/venv/lib/python3.11/site-packages/diffusers/utils/__init__.py)

**Comment by xiaohu2015** [comment]
@5shekel @TheLearner23 hi, you can clone the new repo of https://github.com/tencent-ailab/IP-Adapter

**Comment by 5shekel** [comment]
in windows using virtaul env
```powershell
python -m venv venv
/venv/scripts/activate
pip install diffusers[torch]
```
i get 
```python

>>> from ip_adapter import IPAdapter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\dev\IP-Adapter\ip_adapter\__init__.py", line 1, in <module>
    from .ip_adapter import IPAdapter, IPAdapterXL, IPAdapterPlus
  File "C:\dev\IP-Adapter\ip_adapter\ip_adapter.py", line 6, in <module>
    from diffusers.pipelines.controlnet import MultiControlNetModel
ImportError: cannot import name 'MultiControlNetModel' from 'diffusers.pipelines.controlnet' (C:\dev\IP-Adapter\.venv\lib\site-packages\diffusers\pipelines\controlnet\__init__.py)

```
```
git log
commit d8ab37c421c1ab95d15abe094e8266a6d01e26ef (HEAD -> main, origin/main, origin/HEAD)

pip show diffusers
---
Name: diffusers
Version: 0.21.2
Summary: State-of-the-art diffusion in PyTorch and JAX.
Home-page: https://github.com/huggingface/diffusers
Requires: filelock, huggingface-hub, importlib-metadata, numpy, Pillow, regex, requests, safetensors
```

**Comment by 5shekel** [comment]
changing
`from diffusers.pipelines.controlnet import MultiControlNetModel`
to
`from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion_controlnet import MultiControlNetModel`
seems to be the way (https://github.com/huggingface/diffusers/issues/2556#issuecomment-1518137367)
but need to adapt to other pipelines mentioned in the colab notebooks

`from diffusers import StableDiffusionPipeline, StableDiffusionImg2ImgPipeline, StableDiffusionInpaintPipelineLegacy, DDIMScheduler, AutoencoderKL`

**Comment by Sivll** [comment]
Am I able to run it in macOS terminal?

**Comment by 5shekel** [comment]
You don't Have cuda if you don't have Nvidia card. Maybe this can work in CPU. Idk

**Comment by Sivll** [comment]
There's AMD gpu on board

**Comment by 5shekel** [comment]
update with your mac model so people can assist. i cant, sorry
there is attempt[1] to run on mac new models, but idk about your situation. 
probebly better to open new issue. this is different
[1] https://github.com/tencent-ailab/IP-Adapter/issues/53

**Comment by 5shekel** [comment]
@Sivll try adding this [untested] check to the model loading cell
```python
device = "cuda" if torch.cuda.is_available() else "cpu"
```
instead of
`device = "cude"`

**Comment by Aime-ry** [comment]
To fix this, you can open the file located at /stable-diffusion-webui/extensions/sd-webui-controlnet/annotator/clipvision/init.py using a text editor or IDE (such as VSCode or PyCharm).

1、Navigate to line 81 and locate the line: clip_vision_h_uc = torch.load(clip_vision_h_uc)['uc'].
2、Modify this line to: clip_vision_h_uc = torch.load(clip_vision_h_uc, map_location=torch.device('cpu'))['uc'].
3、Save your changes and exit the editor.
4、Run your program again. This should prevent any CUDA-related errors.

is to add the command-line flag "--no-half" to your launch settings.

**Comment by Clear-3d** [comment]
> https://github.com/tencent-ailab/IP-Adapter/blob/22da45667898fd237ab54d3681db53a9ae98bf1e/ip_adapter/utils.py#L9
> 
> after
> 
> ```python
> pip install diffusers[torch]
> #or
> pip install diffusers
> ```
> 
> im getting
> 
> ```python
> >>> from diffusers.utils import is_compiled_module
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> ImportError: cannot import name 'is_compiled_module' from 'diffusers.utils' (C:\tools\miniconda3\envs\test\lib\site-packages\diffusers\utils\__init__.py)
> ```
> 
> simple `print(dir(diffusers.utils))` cant find it.
> 
> ```python
> >>> import diffusers.utils 
> >>> print(dir(diffusers.utils))
> ['BACKENDS_MAPPING', 'BaseOutput', 'CONFIG_NAME', 'DEPRECATED_REVISION_ARGS', 'DIFFUSERS_CACHE', 'DIFFUSERS_DYNAMIC_MODULE_NAME', 'DummyObject', 'ENV_VARS_TRUE_AND_AUTO_VALUES', 'ENV_VARS_TRUE_VALUES', 'FLAX_WEIGHTS_NAME', 'HF_HUB_OFFLINE', 'HF_MODULES_CACHE', 'HUGGINGFACE_CO_RESOLVE_ENDPOINT', 'ONNX_EXTERNAL_WEIGHTS_NAME', 'ONNX_WEIGHTS_NAME', 'OptionalDependencyNotAvailable', 'PIL_INTERPOLATION', 'PushToHubMixin', 'SAFETENSORS_WEIGHTS_NAME', 'USE_JAX', 'USE_TF', 'USE_TORCH', 'WEIGHTS_NAME', '_LazyModule', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '__version__', '_add_variant', '_get_model_file', 'check_min_version', 'constants', 'deprecate', 'deprecation_utils', 'doc_utils', 'dummy_flax_and_transformers_objects', 'dummy_flax_objects', 'dummy_note_seq_objects', 'dummy_onnx_objects', 'dummy_torch_and_librosa_objects', 'dummy_torch_and_scipy_objects', 'dummy_torch_and_torchsde_objects', 'dummy_torch_and_transformers_and_k_diffusion_objects', 'dummy_torch_and_transformers_and_onnx_objects', 'dummy_torch_and_transformers_objects', 'dummy_transformers_and_torch_and_note_seq_objects', 'dynamic_modules_utils', 'export_to_gif', 'export_to_obj', 'export_to_ply', 'export_to_video', 'export_utils', 'extract_commit_hash', 'get_class_from_dynamic_module', 'get_logger', 'get_objects_from_module', 'http_user_agent', 'hub_utils', 'import_utils', 'is_accelerate_available', 'is_accelerate_version', 'is_bs4_available', 'is_flax_available', 'is_ftfy_available', 'is_inflect_available', 'is_invisible_watermark_available', 'is_k_diffusion_available', 'is_k_diffusion_version', 'is_librosa_available', 'is_note_seq_available', 'is_omegaconf_available', 'is_onnx_available', 'is_scipy_available', 'is_tensorboard_available', 'is_torch_available', 'is_torch_version', 'is_torchsde_available', 'is_transformers_available', 'is_transformers_version', 'is_unidecode_available', 'is_wandb_available', 'is_xformers_available', 'load_image', 'loading_utils', 'logger', 'logging', 'make_image_grid', 'numpy_to_pil', 'os', 'outputs', 'pil_utils', 'pt_to_pil', 'replace_example_docstring', 'requires_backends', 'version']
> ```
> 
> i see others using
> 
> ```python
> from diffusers.utils.torch_utils import is_compiled_module
> ```
I fixed it by using:
from diffusers.utils.torch_utils import is_compiled_module
Thanks!
